### ISSUE TITLE
New Earnings Calendar

### ISSUE THREAD

**Issue by finnhubio** [finnhubio/Finnhub-API#92]
New earnings calendar endpoint with EPS estimate, EPS actual, Revenue estimate and Revenue actual.

[https://finnhub.io/docs/api#earnings-calendar](https://finnhub.io/docs/api#earnings-calendar)

**Comment by bkcollection** [comment]
The earnings calendar is very different from the website from Nasdaq
https://www.nasdaq.com/market-activity/earnings
your earnings calendar is late.

**Comment by bkcollection** [comment]
Besides, I suggest the earning calendar can be separated base on exchanges, such as US, HK or etc

**Comment by mrfroasty** [comment]
I totally agree that is much likely handy to have exchange default US (optional argument).

**Comment by mrfroasty** [comment]
> The earnings calendar is very different from the website from Nasdaq
https://www.nasdaq.com/market-activity/earnings
your earnings calendar is late.
By the way you can simply add from && to arguments, the earnings of other exchanges I decided to filter them off with only in the exchange I am dealing with.

**Comment by mrfroasty** [comment]
The calendar gives 500 items of symbols completely irrelevant if someone is intending to fetch all US exchange. This forces user to start calling symbols only one by one instead leaving symbol out to get all data.

**Comment by finnhubio** [comment]
We changed the default to US markets only. You can pass in the param &international=true to get data for international markets. See docs for more information.

**Comment by d3an** [comment]
I'm trying to get a list of all the historical earnings dates (with the hour response parameter) for a stock. I'm currently on the  free plan, but the existing API only shows me 2020 dates or maybe the first quarter of 2019. 

Can you provide a demo request and response to this?

**Comment by finnhubio** [comment]
> I'm trying to get a list of all the historical earnings dates (with the hour response parameter) for a stock. I'm currently on the free plan, but the existing API only shows me 2020 dates or maybe the first quarter of 2019.
> 
> Can you provide a demo request and response to this?

You only get 1 year of historical earnings with the free plan. You can upgrade your plan here: https://finnhub.io/pricing-stock-estimates

**Comment by d3an** [comment]
> > I'm trying to get a list of all the historical earnings dates (with the hour response parameter) for a stock. I'm currently on the free plan, but the existing API only shows me 2020 dates or maybe the first quarter of 2019.
> > Can you provide a demo request and response to this?
> 
> You only get 1 year of historical earnings with the free plan. You can upgrade your plan here: https://finnhub.io/pricing-stock-estimates

Can you give a sample request and response of what the full version gives? I'd be interested in the response showing all historical earnings dates for a stock of your choice.

**Comment by finnhubio** [comment]
> > > I'm trying to get a list of all the historical earnings dates (with the hour response parameter) for a stock. I'm currently on the free plan, but the existing API only shows me 2020 dates or maybe the first quarter of 2019.
> > > Can you provide a demo request and response to this?
> > 
> > 
> > You only get 1 year of historical earnings with the free plan. You can upgrade your plan here: https://finnhub.io/pricing-stock-estimates
> 
> Can you give a sample request and response of what the full version gives? I'd be interested in the response showing all historical earnings dates for a stock of your choice.

It will give you the exact same date structure with your plan but with more data. You can try this request: https://finnhub.io/api/v1/calendar/earnings?from=2010-01-01&to=2020-07-29&symbol=AAPL&token=
